### PR TITLE
Kam changes

### PR DESCRIPTION
**Pull request by Sir-Kam**
Lots of changes, to start with at least.

### CODE REVIEW

**Comment by Sir-Kam** [comment]
Just check it over for stuff. @Draycia

File: src/ComponentManager.ts [hunk 1]
```diff
@@ -45,78 +98,113 @@ export default class ModuleLoader {
...
   private onBot() {
     const fs = require('fs');
-
-    this.modules["ComponentManager.ts"] = this.cmCommands;
+    this.modules2.Add("ComponentManager", this.managerCommands);
+    this.iModules.Add("ComponentManager", new KeyValueArray<Command>());
+    // console.log("- ComponentManager")
+    for(let mod in this.managerCommands) {
+      // console.log(mod)
+      this.iModules.Item("ComponentManager").Add(mod, this.oldparamss[mod]);
```
Comment (Draycia) [inline src/ComponentManager.ts]: `oldparamms` doesn't seem to exist anywhere else in the file, and the compiler is whining about this too.

File: src/Modules/Helper.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import Discord = require('discord.js');
+import Command from './Command';
+import KeyValueArray, { Cmd } from './KeyValue';
```
Comment (Draycia) [inline src/Modules/Helper.ts]: KeyValue.ts doesn't seem to have anything called `Cmd` and the compiler is complaining about this. There is `Command` in it, though.

**Comment by Sir-Kam** [comment]
Alright. Changes made

**Pull request merged**